test/use_case/video_search/VideoSearchInteractorTest.java

```diff
@@ -0,0 +1,79 @@
+package use_case.video_search;
+
+import data_access.InMemoryHistoryDataAccessObject;
+import data_access.VideoSearchDataAccessObject;
+import data_access.YouTubeDataAccess;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class VideoSearchInteractorTest {
+
+    @BeforeEach
+    void setUp() {
+    }
+
+    @Test
+    void successTest() throws IOException, GeneralSecurityException {
+
+        // input data + data access interface
+        VideoSearchInputData inputData = new VideoSearchInputData("H-v6DfxnjF8", "user");
+        VideoSearchDataAccessInterface youtubeRepository = new VideoSearchDataAccessObject();
+
+        VideoSearchUserDataAccessInterface historyRepository = new InMemoryHistoryDataAccessObject();
+
+        // presenter + success and fail view
+        VideoSearchOutputBoundary presenter = new VideoSearchOutputBoundary() {
+            @Override
+            public void prepareSuccessView(VideoSearchOutputData video) {
+                assertEquals("H-v6DfxnjF8", video.getVideoId());
+                assertEquals("FIE Fencing Channel", video.getChannelName());
+                assertEquals("Grand Prix Turin 2020 Women's Foil Final THIBUS FRA vs KIEFER USA", video.getTitle());
+                assertEquals("", video.getDescription());
+//                assertEquals("", video.getVideoPublishDate());
+                assertEquals(12996, video.getViewCount());
+                assertEquals(169, video.getLikeCount());
+                assertEquals(9, video.getCommentCount());
+
+            }
+
+            @Override
+            public void prepareFailView(String error) {
+                fail("This use case should not fail.");
+            }
+
+        };
+
+        // interactor + execute
+        VideoSearchInputBoundary interactor = new VideoSearchInteractor(youtubeRepository, historyRepository, presenter);
+        interactor.execute(inputData);
```

Comment:
Does this account for the video growth in statistics like view count, like count, and comment count?